wasm/src/main/scala/ir2wasm/HelperFunctions.scala

```diff
@@ -1349,6 +1350,136 @@ object HelperFunctions {
     fctx.buildAndAddToContext()
   }
 
+  /** `identityHashCode`: `anyref -> i32`.
+    *
+    * This is the implementation of `IdentityHashCode`. It is also used to compute the `hashCode()`
+    * of primitive values when dispatch is required, so it must be consistent with the
```

Comment:
What do you mean by `dispatch is required`?
IIUC, this `identityHashCode` is used for the primitive values (other than `long` and `char`
where the receiver (of `hashCode`) type couldn't known as primitive types at compile-time (and caught at these branches https://github.com/tanishiking/scala-wasm/blob/5b184de8ea4e2f69d813614438f12e36f28ddd88/wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala#L292-L305 )

Reply:
Ah, ok, "dispatch is **not** required" when the `t.receiver.tpe` is known as `PrimType` or a `HIjackedClass` at compile-time, because they will be translated to `ApplyStatically` and we don't need to vtable/itable-dispatch in those cases